docs/alerts.md

```diff
@@ -0,0 +1,264 @@
+|20|30|notify|crit|true|cpu_usage|pagerduty|
+|30|40|notify|crit|false|cpu_usage|pagerduty|
+|40|50|notify|warn|true|cpu_usage|pagerduty|
+|50|60|notify|crit|true|cpu_usage|pagerduty|
+|60|70|notify|warn|true|cpu_usage|pagerduty|
+|70|80|notify|warn|false|cpu_usage|pagerduty|
+|120|130|resolve|ok|true|cpu_usage|pagerduty|
+
+As can be noticed, the handler did not do anything with the `unknown` levels and so there was nothing to log.
+The `_at` time is also 10 seconds after the time the event occured.
+This is because it is the time the action was _performed_ rather than when the check happened.
+This assumes that the table is triggered upon receiving the next point in the sequence so it is triggered
+on receiving the next point after 10 seconds.
+
+If we were in a situation where the handler were to skip an action because a future notification pre-empted it (such as if only a single table were received and the handler noticed that the alert was already resolved), the `_at` time would be null since the action never occurred, but the row would still be present.
+
+#### Additional Notes
+
+As can be seen, the `notify()` method is simply a placeholder to allow notification handlers to be abstracted from the notification itself.
+This lends itself better to code generators because the `notify()` method can be customized using extern properties so that the act of checking and notifying can be separated from eachother.
+
+## Integration Methods
+
+The above constructs lend themselves well to various different methods of integration and also allow for building out future integration plans.
+
+### Inline
+
+It is possible for the entire query to run inline.
+
+```
+// Send to multiple notification handlers.
+notify = (tables=<-) => {
+	tables |> pagerduty.notify(routingKey: myRoutingKey)
+	tables |> slack.notify(channel: "#flux")
+}
+
+from(bucket: "telegraf")
+    |> range(start: -5m)
+    |> filter(fn: (r) => r._measurement == "cpu" and r._field == "usage_system")
+    |> check(
+    	crit: (r) => r._value > 90,
+    )
+    |> notify()
+```
+
+### Using Buckets
+
+Using the same script as the above, the `notify` method can be customized to write to a bucket instead and the notifications can be triggered off of the contents of the bucket instead.
+
+```
+// This goes in the check script.
+notify = (tables=<-) => tables |> to(bucket: "system")
+
+// This goes in a separate script.
+alert = (tables=<-, handler) => tables |> handler() |> to(bucket: "notifications")
+
+tables = from(bucket: "system")
+	|> range(start: -1m)
+	|> notify()
+	
+// The function is easy for the user to write and it is easy to
+// inject into a generated script.
+handler1 = (tables=<-) => pagerduty.notify(routingKey: myRoutingKey)
+handler2 = (tables=<-) => slack.notify(channel: "#flux")
+	
+// Send to multiple notification handlers.
+// This is easy to generate.
+tables |> handler1() |> to(bucket: "notifications")
+tables |> handler2() |> to(bucket: "notifications")
```

Comment:
This example is a bit confusing, where it the `check` in this example? Or is this just and example that is reading from the `check status` system bucket? Maybe it would help to write out the separate tasks completely and show how they interact?

Reply:
This is an example that reads from the `check status` system bucket. Or `tables` is defined in some other way.

I do think it might be helpful to write separate tasks entirely. I'll work on that.